scripts/dependencies/from_package.py

```diff
@@ -18,19 +18,24 @@
 import re
 import sys
 
-pattern = re.compile(u'^{}'.format(sys.argv[1]))
+package_path = sys.argv[1]
+if package_path.endswith('/'):
+    package_path = package_path[:-1]
+
+pattern = re.compile(u'{}'.format(package_path))
 
 data_path = 'reports/dependencies/dependencies.txt'
 if not os.path.exists(data_path):
     print('The dependencies data file is unavailable; run scripts/dependencies/enumerate.sh first.')
+
 with open(data_path, 'r') as f:
     for dep in map(eval, f):
         (from_root, from_name), (to_root, to_name) = dep
         if to_name is None:
             continue
-        if pattern.search(from_name) and not pattern.search(to_name):
+        if pattern.search(from_root) and not pattern.search(to_root):
```

Comment:
The root values are typically "/edx/app/edxapp/edx-platform" or "/edx/app/edxapp/venvs/edxapp/lib/python2.7/site-packages", i.e. the directory in which the package was found.  The names are the paths to each file starting with the package, such as "openedx/features/course_experience/views/course_updates.py".  So I'm pretty sure it's the latter we want to search.

Reply:
Ah, that's because that isn't a valid package name.  You'd want `scripts/dependencies/from_package.py common/test` for code under `/edx/app/edxapp/edx-platform/common/test`, `scripts/dependencies/from_package.py calc` for code under `/edx/app/edxapp/edx-platform/common/lib/calc`, etc.  The script works on packages as imported in python code, not directory paths.

Reply:
That seems reasonable but the results weren't what I expected.  On master:
```
root@lms:/edx/app/edxapp/edx-platform# scripts/dependencies/from_package.py common/djangoapps | wc -l
0
```

On this branch:
```
root@lms:/edx/app/edxapp/edx-platform# scripts/dependencies/from_package.py common/djangoapps | wc -l
390
```

Reply:
`common/djangoapps` also isn't a package; the individual directories under it are.  The layout of packages in edx-platform is unfortunately a bit of a mess.

Reply:
gotcha, I think that's were my confusion is stemming from.  I'll probably close this for now then.